app/models/teacher_set.rb

```diff
@@ -198,6 +198,8 @@ def self.create_or_update_teacher_set(req_body)
       # If bib request-body has suppressed value as true then delete teacher-set from database and elastic-search.
       teacher_set = self.delete_teacher_set(bib_id, req_body["suppressed"])
       if teacher_set.destroyed?
+        LogWrapper.log('INFO', {message: "message: #{BIB_RECORD_SUPPRESSED_REMOVED_FROM_MLN[:msg]}",
+                       method: __method__, bib_id: bib_id, teacher_set_id: teacher_set.id, suppressed: req_body["suppressed"]})
```

Comment:
I assume `teacher_set.id` still has a value even after it's been deleted? Just wondering if you needed to save `ts_id` before doing `delete_teacher_set`

Reply:
@thisisstephenbetts  self.delete_teacher_set(bib_id, req_body["suppressed"]) returns teacher_set object. I already saved teacher_set object before deletion.

``` # Delete teacher-set record from db and elastic search.
  def self.delete_teacher_set(bib_id, suppressed=false)
    # Get teacher-set record by bib_id
    teacher_set = self.get_teacher_set_by_bnumber(bib_id)
    unless teacher_set.present?
      if suppressed
        raise SuppressedBibRecordException.new(BIB_RECORD_SUPPRESSED_NOT_ADDED_TO_MLN[:code],BIB_RECORD_SUPPRESSED_NOT_ADDED_TO_MLN[:msg])
      else
        raise BibRecordNotFoundException.new(BIB_RECORD_NOT_FOUND[:code],BIB_RECORD_NOT_FOUND[:msg])
      end
    end
    
    # Delete teacher-set record
    resp = teacher_set.destroy
    # Feature flag: 'teacherset.data.from.elasticsearch.enabled = true'.
    # If feature flag is enabled delete data from elasticsearch.
    if MlnConfigurationController.new.feature_flag_config('teacherset.data.from.elasticsearch.enabled')
      # After deletion of teacherset data from db than delete teacherset doc from elastic search
      teacher_set.delete_teacherset_record_from_es if resp.destroyed?
    end
    teacher_set
  end```

Reply:
Great!